src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp

```diff
@@ -226,59 +182,60 @@ static boost::filesystem::path findXilinxRootPort(void)
                     file.close();
                 }
             }
-
             catch (const boost::filesystem::ifstream::failure& e)
             {
-                continue; 
+                continue;
             }
-        
-            if (!vendor_id.compare(XILINX_VENDOR) && !device_id.compare(XILINX_US))
-                return rootPortPath;
-        }
 
-        rootPortPath = "";
+            if (!vendor_id.compare(XILINX_VENDOR) && !device_id.compare(XILINX_US)) 
+                if (!check_mgmt_present(iPath)) 
+                    rootportList.push_back(iPath.string());
+        }
     }
 
-    return rootPortPath;
+    return rootportList;
 }
 
 static int hotplugRescan(void)
 {
-    boost::filesystem::ofstream ofile;     
-    boost::filesystem::path sysfs_path = findXilinxRootPort();
+    boost::filesystem::ofstream ofile;
+    /* Find all the RootPorts where Xilinx devices are connected */
+    std::list<std::string> rootportList = findXilinxRootPort(SYSFS_PATH);
```

Comment:
i'm not sure why not just rescan from the highest level? if you remove the board, do you still see xilinx's pcie bridge?